File: binding/src/libqpushbutton.cpp
#include <QAbstractButton>
#include <QEvent>
#include <QFocusEvent>
#include <QIcon>
#include <QKeyEvent>
#include <QMenu>
#include <QMouseEvent>
#include <QPaintEvent>
#include <QPoint>
#include <QPushButton>
#include <QSize>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QStyleOptionButton>
#include <QWidget>
#include <qpushbutton.h>
#include "libqpushbutton.h"
#include "libqpushbutton.hxx"

QPushButton* QPushButton_new(QWidget* parent) {
	 return new VirtualQPushButton(parent);
}

QPushButton* QPushButton_new2() {
	 return new VirtualQPushButton();
}

QPushButton* QPushButton_new3(const libqt_string text) {
	 return new VirtualQPushButton(QString::fromUtf8(text.data, text.len));
}

QPushButton* QPushButton_new4(const QIcon* icon, const libqt_string text) {
	 return new VirtualQPushButton(*icon, QString::fromUtf8(text.data, text.len));
}

QPushButton* QPushButton_new5(const libqt_string text, QWidget* parent) {
	 return new VirtualQPushButton(QString::fromUtf8(text.data, text.len), parent);
}

QPushButton* QPushButton_new6(const QIcon* icon, const libqt_string text, QWidget* parent) {
	 return new VirtualQPushButton(*icon, QString::fromUtf8(text.data, text.len), parent);
}

libqt_string QPushButton_Tr(const char* s) {
	QString _ret = QPushButton::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QSize* QPushButton_SizeHint(const QPushButton* self) {
	return new QSize(self->sizeHint());
}

QSize* QPushButton_MinimumSizeHint(const QPushButton* self) {
	return new QSize(self->minimumSizeHint());
}

bool QPushButton_AutoDefault(const QPushButton* self) {
	return self->autoDefault();
}

void QPushButton_SetAutoDefault(QPushButton* self, bool autoDefault) {
	self->setAutoDefault(autoDefault);
}

bool QPushButton_IsDefault(const QPushButton* self) {
	return self->isDefault();
}

void QPushButton_SetDefault(QPushButton* self, bool defaultVal) {
	self->setDefault(defaultVal);
}

void QPushButton_SetMenu(QPushButton* self, QMenu* menu) {
	self->setMenu(menu);
}

QMenu* QPushButton_Menu(const QPushButton* self) {
	return self->menu();
}

void QPushButton_SetFlat(QPushButton* self, bool flat) {
	self->setFlat(flat);
}

bool QPushButton_IsFlat(const QPushButton* self) {
	return self->isFlat();
}

void QPushButton_ShowMenu(QPushButton* self) {
	self->showMenu();
}

libqt_string QPushButton_Tr2(const char* s, const char* c) {
	QString _ret = QPushButton::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QPushButton_Tr3(const char* s, const char* c, int n) {
	QString _ret = QPushButton::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
QSize* QPushButton_QBaseSizeHint(const QPushButton* self) {
	auto* vqpushbutton = dynamic_cast<const VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_SizeHint_IsBase(true);
	return new QSize(vqpushbutton->sizeHint());
}
}

// Auxiliary method to allow providing re-implementation
void QPushButton_OnSizeHint(const QPushButton* self, intptr_t slot) {
	auto* vqpushbutton = dynamic_cast<const VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_SizeHint_Callback(reinterpret_cast<VirtualQPushButton::QPushButton_SizeHint_Callback>(slot));
}
}

// Base class handler implementation
QSize* QPushButton_QBaseMinimumSizeHint(const QPushButton* self) {
	auto* vqpushbutton = dynamic_cast<const VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_MinimumSizeHint_IsBase(true);
	return new QSize(vqpushbutton->minimumSizeHint());
}
}

// Auxiliary method to allow providing re-implementation
void QPushButton_OnMinimumSizeHint(const QPushButton* self, intptr_t slot) {
	auto* vqpushbutton = dynamic_cast<const VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_MinimumSizeHint_Callback(reinterpret_cast<VirtualQPushButton::QPushButton_MinimumSizeHint_Callback>(slot));
}
}

// Derived class handler implementation
bool QPushButton_Event(QPushButton* self, QEvent* e) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
	return vqpushbutton->event(e);
	} else {
	return self->QPushButton::event(e);
}
}

// Base class handler implementation
bool QPushButton_QBaseEvent(QPushButton* self, QEvent* e) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_Event_IsBase(true);
	return vqpushbutton->event(e);
}
}

// Auxiliary method to allow providing re-implementation
void QPushButton_OnEvent(QPushButton* self, intptr_t slot) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_Event_Callback(reinterpret_cast<VirtualQPushButton::QPushButton_Event_Callback>(slot));
}
}

// Derived class handler implementation
void QPushButton_PaintEvent(QPushButton* self, QPaintEvent* param1) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
	vqpushbutton->paintEvent(param1);
	} else {
	self->QPushButton::paintEvent(param1);
}
}

// Base class handler implementation
void QPushButton_QBasePaintEvent(QPushButton* self, QPaintEvent* param1) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_PaintEvent_IsBase(true);
	vqpushbutton->paintEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QPushButton_OnPaintEvent(QPushButton* self, intptr_t slot) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_PaintEvent_Callback(reinterpret_cast<VirtualQPushButton::QPushButton_PaintEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QPushButton_KeyPressEvent(QPushButton* self, QKeyEvent* param1) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
	vqpushbutton->keyPressEvent(param1);
	} else {
	self->QPushButton::keyPressEvent(param1);
}
}

// Base class handler implementation
void QPushButton_QBaseKeyPressEvent(QPushButton* self, QKeyEvent* param1) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_KeyPressEvent_IsBase(true);
	vqpushbutton->keyPressEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QPushButton_OnKeyPressEvent(QPushButton* self, intptr_t slot) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_KeyPressEvent_Callback(reinterpret_cast<VirtualQPushButton::QPushButton_KeyPressEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QPushButton_FocusInEvent(QPushButton* self, QFocusEvent* param1) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
	vqpushbutton->focusInEvent(param1);
	} else {
	self->QPushButton::focusInEvent(param1);
}
}

// Base class handler implementation
void QPushButton_QBaseFocusInEvent(QPushButton* self, QFocusEvent* param1) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_FocusInEvent_IsBase(true);
	vqpushbutton->focusInEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QPushButton_OnFocusInEvent(QPushButton* self, intptr_t slot) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_FocusInEvent_Callback(reinterpret_cast<VirtualQPushButton::QPushButton_FocusInEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QPushButton_FocusOutEvent(QPushButton* self, QFocusEvent* param1) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
	vqpushbutton->focusOutEvent(param1);
	} else {
	self->QPushButton::focusOutEvent(param1);
}
}

// Base class handler implementation
void QPushButton_QBaseFocusOutEvent(QPushButton* self, QFocusEvent* param1) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_FocusOutEvent_IsBase(true);
	vqpushbutton->focusOutEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QPushButton_OnFocusOutEvent(QPushButton* self, intptr_t slot) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_FocusOutEvent_Callback(reinterpret_cast<VirtualQPushButton::QPushButton_FocusOutEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QPushButton_MouseMoveEvent(QPushButton* self, QMouseEvent* param1) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
	vqpushbutton->mouseMoveEvent(param1);
	} else {
	self->QPushButton::mouseMoveEvent(param1);
}
}

// Base class handler implementation
void QPushButton_QBaseMouseMoveEvent(QPushButton* self, QMouseEvent* param1) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_MouseMoveEvent_IsBase(true);
	vqpushbutton->mouseMoveEvent(param1);
}
}

// Auxiliary method to allow providing re-implementation
void QPushButton_OnMouseMoveEvent(QPushButton* self, intptr_t slot) {
	auto* vqpushbutton = dynamic_cast<VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_MouseMoveEvent_Callback(reinterpret_cast<VirtualQPushButton::QPushButton_MouseMoveEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QPushButton_InitStyleOption(const QPushButton* self, QStyleOptionButton* option) {
	auto* vqpushbutton = dynamic_cast<const VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
	vqpushbutton->initStyleOption(option);
	} else {
	self->QPushButton::initStyleOption(option);
}
}

// Base class handler implementation
void QPushButton_QBaseInitStyleOption(const QPushButton* self, QStyleOptionButton* option) {
	auto* vqpushbutton = dynamic_cast<const VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_InitStyleOption_IsBase(true);
	vqpushbutton->initStyleOption(option);
}
}

// Auxiliary method to allow providing re-implementation
void QPushButton_OnInitStyleOption(const QPushButton* self, intptr_t slot) {
	auto* vqpushbutton = dynamic_cast<const VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_InitStyleOption_Callback(reinterpret_cast<VirtualQPushButton::QPushButton_InitStyleOption_Callback>(slot));
}
}

// Derived class handler implementation
bool QPushButton_HitButton(const QPushButton* self, const QPoint* pos) {
	auto* vqpushbutton = dynamic_cast<const VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
	return vqpushbutton->hitButton(*pos);
	} else {
	return self->QPushButton::hitButton(*pos);
}
}

// Base class handler implementation
bool QPushButton_QBaseHitButton(const QPushButton* self, const QPoint* pos) {
	auto* vqpushbutton = dynamic_cast<const VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_HitButton_IsBase(true);
	return vqpushbutton->hitButton(*pos);
}
}

// Auxiliary method to allow providing re-implementation
void QPushButton_OnHitButton(const QPushButton* self, intptr_t slot) {
	auto* vqpushbutton = dynamic_cast<const VirtualQPushButton*>(self);
	if (vqpushbutton && vqpushbutton->isVirtualQPushButton) {
vqpushbutton->setQPushButton_HitButton_Callback(reinterpret_cast<VirtualQPushButton::QPushButton_HitButton_Callback>(slot));
}
}

void QPushButton_Delete(QPushButton* self) {
    delete self;
}

File: binding/include/libqpushbutton.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qpushbutton.html
pub const qpushbutton = struct {

    /// New constructs a new QPushButton object.
    pub fn New(parent: ?*anyopaque) QtC.QPushButton {
        return qtc.QPushButton_new(@ptrCast(parent));
    }


    /// New2 constructs a new QPushButton object.
    pub fn New2() QtC.QPushButton {
        return qtc.QPushButton_new2();
    }


    /// New3 constructs a new QPushButton object.
    pub fn New3(text: []const u8) QtC.QPushButton {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QPushButton_new3(text_str);
    }


    /// New4 constructs a new QPushButton object.
    pub fn New4(icon: ?*anyopaque, text: []const u8) QtC.QPushButton {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QPushButton_new4(@ptrCast(icon), text_str);
    }


    /// New5 constructs a new QPushButton object.
    pub fn New5(text: []const u8, parent: ?*anyopaque) QtC.QPushButton {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QPushButton_new5(text_str, @ptrCast(parent));
    }


    /// New6 constructs a new QPushButton object.
    pub fn New6(icon: ?*anyopaque, text: []const u8, parent: ?*anyopaque) QtC.QPushButton {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
return qtc.QPushButton_new6(@ptrCast(icon), text_str, @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPushButton_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpushbutton.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QPushButton_SizeHint(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QSize) void {
        qtc.QPushButton_OnSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QPushButton_QBaseSizeHint(@ptrCast(self));
    }

    pub fn MinimumSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QPushButton_MinimumSizeHint(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnMinimumSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QSize) void {
        qtc.QPushButton_OnMinimumSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMinimumSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QPushButton_QBaseMinimumSizeHint(@ptrCast(self));
    }

    pub fn AutoDefault(self: ?*anyopaque, ) bool {
        return qtc.QPushButton_AutoDefault(@ptrCast(self));
    }

    pub fn SetAutoDefault(self: ?*anyopaque, autoDefault: bool) void {
        qtc.QPushButton_SetAutoDefault(@ptrCast(self), autoDefault);
    }

    pub fn IsDefault(self: ?*anyopaque, ) bool {
        return qtc.QPushButton_IsDefault(@ptrCast(self));
    }

    pub fn SetDefault(self: ?*anyopaque, defaultVal: bool) void {
        qtc.QPushButton_SetDefault(@ptrCast(self), defaultVal);
    }

    pub fn SetMenu(self: ?*anyopaque, menu: ?*anyopaque) void {
        qtc.QPushButton_SetMenu(@ptrCast(self), @ptrCast(menu));
    }

    pub fn Menu(self: ?*anyopaque, ) QtC.QMenu {
        return qtc.QPushButton_Menu(@ptrCast(self));
    }

    pub fn SetFlat(self: ?*anyopaque, flat: bool) void {
        qtc.QPushButton_SetFlat(@ptrCast(self), flat);
    }

    pub fn IsFlat(self: ?*anyopaque, ) bool {
        return qtc.QPushButton_IsFlat(@ptrCast(self));
    }

    pub fn ShowMenu(self: ?*anyopaque, ) void {
        qtc.QPushButton_ShowMenu(@ptrCast(self));
    }

    pub fn Event(self: ?*anyopaque, e: ?*anyopaque) bool {
        return qtc.QPushButton_Event(@ptrCast(self), @ptrCast(e));
    }

    /// Allows for overriding the related default method
    pub fn OnEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QPushButton_OnEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseEvent(self: ?*anyopaque, e: ?*anyopaque) bool {
        return qtc.QPushButton_QBaseEvent(@ptrCast(self), @ptrCast(e));
    }

    pub fn PaintEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QPushButton_PaintEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnPaintEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QPushButton_OnPaintEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBasePaintEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QPushButton_QBasePaintEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn KeyPressEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QPushButton_KeyPressEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnKeyPressEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QPushButton_OnKeyPressEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseKeyPressEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QPushButton_QBaseKeyPressEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn FocusInEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QPushButton_FocusInEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnFocusInEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QPushButton_OnFocusInEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseFocusInEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QPushButton_QBaseFocusInEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn FocusOutEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QPushButton_FocusOutEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnFocusOutEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QPushButton_OnFocusOutEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseFocusOutEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QPushButton_QBaseFocusOutEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn MouseMoveEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QPushButton_MouseMoveEvent(@ptrCast(self), @ptrCast(param1));
    }

    /// Allows for overriding the related default method
    pub fn OnMouseMoveEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QPushButton_OnMouseMoveEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMouseMoveEvent(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.QPushButton_QBaseMouseMoveEvent(@ptrCast(self), @ptrCast(param1));
    }

    pub fn InitStyleOption(self: ?*anyopaque, option: ?*anyopaque) void {
        qtc.QPushButton_InitStyleOption(@ptrCast(self), @ptrCast(option));
    }

    /// Allows for overriding the related default method
    pub fn OnInitStyleOption(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QPushButton_OnInitStyleOption(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseInitStyleOption(self: ?*anyopaque, option: ?*anyopaque) void {
        qtc.QPushButton_QBaseInitStyleOption(@ptrCast(self), @ptrCast(option));
    }

    pub fn HitButton(self: ?*anyopaque, pos: ?*anyopaque) bool {
        return qtc.QPushButton_HitButton(@ptrCast(self), @ptrCast(pos));
    }

    /// Allows for overriding the related default method
    pub fn OnHitButton(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QPushButton_OnHitButton(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseHitButton(self: ?*anyopaque, pos: ?*anyopaque) bool {
        return qtc.QPushButton_QBaseHitButton(@ptrCast(self), @ptrCast(pos));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPushButton_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpushbutton.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPushButton_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpushbutton.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPushButton_Delete(@ptrCast(self));
    }
};
